patches/spec.fetched.json/08-make-vrf-required-param-in-vrf-metal-gateway.patch

```diff
@@ -1,10 +1,69 @@
+diff --git a/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml b/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml
+new file mode 100644
+index 0000000..42bb973
+--- /dev/null
++++ b/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml
+@@ -0,0 +1,3 @@
++anyOf:
++- $ref: './IPReservation.yaml'
++- $ref: './Href.yaml'
+diff --git a/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml b/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
+index ed0eb30..64aa9a5 100644
+--- a/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
++++ b/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
+@@ -10,7 +10,7 @@ properties:
+     format: uuid
+     type: string
+   ip_reservation:
+-    $ref: './IPReservation.yaml'
++    $ref: './IPReservationOrHref.yaml'
```

Comment:
Does this introduce the same problem we observed with Metal Gateways and VRF Metal Gateways?  Namely: the `Href` model comes first alphabetically, so if the json for an IP reservation includes the `href` property, it'll validate against the `Href` model and you won't be able to get the `IPReservation` instance.

Reply:
yes, it might as Href response will also will be valid for any response that contains href attribute but can be resolved using `additionalProperties: false` and should fix it irrespective of naming order in which it gets deserialised.

Reply:
Based on offline conversation you're OK with accepting that behavior the validation behavior described in my comment for now, is that correct?

Reply:
I'm making minimum changes, until we have the correct set of guidelines to handle such models. I dont want to add `additionalProperties: false` in Href yaml as it might not be pattern we will follow later down the line. And for now its acceptable behaviour described in your above comment as for now we dont need Ip reservation info from the said API.